pkg/sdn/plugin/node.go

```diff
@@ -63,6 +78,10 @@ func NewNodePlugin(pluginName string, osClient *osclient.Client, kClient *kclien
 		}
 	}
 
+	if err := writeCNIConfig(masterKubeConfig, hostname, mtu); err != nil {
```

Comment:
/etc is for static configuration. If the options can really only be determined at run time, then the file shouldn't be written to /etc, it should be written to /run (which I guess means CNI needs to be updated).